src/friendly-iframe-embed.js

```diff
@@ -758,84 +785,103 @@ export class FriendlyIframeEmbed {
     const parentWin = toWin(childWin.frameElement.ownerDocument.defaultView);
     setParentWindow(childWin, parentWin);
 
-    // Install necessary polyfills.
-    installPolyfillsInChildWindow(parentWin, childWin);
-
-    // Install runtime styles.
-    installStylesLegacy(
-      childWin.document,
-      ampSharedCss,
-      /* callback */ null,
-      /* opt_isRuntimeCss */ true,
-      /* opt_ext */ 'amp-runtime'
-    );
-
-    // Run pre-install callback.
-    if (opt_preinstallCallback) {
-      opt_preinstallCallback(childWin);
-    }
-
-    // Install embeddable standard services.
-    installStandardServicesInEmbed(childWin);
-
-    // Install built-ins and legacy elements.
-    copyBuiltinElementsToChildWindow(topWin, childWin);
-    stubLegacyElements(childWin);
-
-    const promises = [];
     extensionIds.forEach((extensionId) => {
       // This will extend automatic upgrade of custom elements from top
       // window to the child window.
       if (!LEGACY_ELEMENTS.includes(extensionId)) {
         stubElementIfNotKnown(childWin, extensionId);
       }
+    });
 
-      // Install CSS.
-      const promise = extensions
-        .preloadExtension(extensionId)
-        .then((extension) => {
-          // Adopt embeddable extension services.
-          extension.services.forEach((service) => {
-            installServiceInEmbedIfEmbeddable(childWin, service.serviceClass);
-          });
-
-          // Adopt the custom elements.
-          let elementPromises = null;
-          for (const elementName in extension.elements) {
-            const elementDef = extension.elements[elementName];
-            const elementPromise = new Promise((resolve) => {
-              if (elementDef.css) {
-                installStylesLegacy(
-                  childWin.document,
-                  elementDef.css,
-                  /* completeCallback */ resolve,
-                  /* isRuntime */ false,
-                  extensionId
+    const maybeDelay = getDelayPromise(parentWin);
+    return Promise.resolve()
+      .then(maybeDelay)
+      .then(() => {
+        // Install necessary polyfills.
+        installPolyfillsInChildWindow(parentWin, childWin);
+      })
+      .then(maybeDelay)
```

Comment:
What's the value of adding an empty microtask here and all below?

Reply:
For the experiment group, maybeDelay will be macrotasks (`setTimeout`). They are intended to cause more chunking.

Reply:
I see. We should change the name to make it clear that it's function that returns a new promise. Maybe `maybeGetDelay`?